Don't expose an outline subsymbol for SVG fill symbol layers

nyalldawson opened the pull request:
This is inconsistent with all other fill types, which don't
have a special outline subsymbol. Users should be creating
outline layers when they want an outline instead.

When reading an SVG fill symbol layer from XML, automatically
upgrade any outline subsymbol which is used by the fill to
be separate symbol layers for the parent fill symbol so
that existing symbols will appear as designed in older QGIS
versions.

Fixes #14103

nyalldawson committed:
Update test, don't assume that SVG fills have a subsymbol by default

nyalldawson committed:
Fix some non-error debug msgs being shown at level 1

nirvn reviewed (approved):
Very, _very_ good move. Thanks

nyalldawson committed:
Don't expose an outline subsymbol for SVG fill symbol layers

This is inconsistent with all other fill types, which don't
have a special outline subsymbol. Users should be creating
outline layers when they want an outline instead.

When reading an SVG fill symbol layer from XML, automatically
upgrade any outline subsymbol which is used by the fill to
be separate symbol layers for the parent fill symbol so
that existing symbols will appear as designed in older QGIS
versions.

Fixes qgis#14103
